pkg/ebpf/c/tracee.bpf.c

```diff
@@ -4748,6 +4750,44 @@ int BPF_KPROBE(trace_security_inode_mknod)
     return events_perf_submit(&data, SECURITY_INODE_MKNOD, 0);
 }
 
+SEC("kprobe/__register_chrdev")
+TRACE_ENT_FUNC(__register_chrdev, REGISTER_CHRDEV);
+
+SEC("kretprobe/__register_chrdev")
+int BPF_KPROBE(trace_ret__register_chrdev)
+{
+    args_t saved_args;
+    if (load_args(&saved_args, REGISTER_CHRDEV) != 0) {
+        // missed entry or not traced
+        return 0;
+    }
+
+    event_data_t data = {};
+    if (!init_event_data(&data, ctx))
+        return 0;
+
+    if (!should_trace(&data.context))
+        return 0;
+
+    unsigned int major_number = (unsigned int)saved_args.args[0];
+    unsigned int returned_major = PT_REGS_RC(ctx);
+
+    // sets the returned major to the requested one in case of a successful registration
+    if (major_number > 0 && returned_major == 0)
+    {
+        returned_major = major_number;
+    }
+
+    char* char_device_name = (char*)saved_args.args[3];
+    struct file_operations *char_device_fops = (struct file_operations *)saved_args.args[4];
+
+    save_to_submit_buf(&data, &major_number, sizeof(unsigned int), 0);
+    save_to_submit_buf(&data, &returned_major, sizeof(unsigned int), 1);
+    save_str_to_buf(&data, char_device_name, 2);
+    save_to_submit_buf(&data, &char_device_fops, sizeof(void *), 3);
```

Comment:
I asked Itamar if he was parsing those pointer arguments into hex so they could be easily consumable in userland. I even opened an issue (FYIO): https://github.com/aquasecurity/tracee/issues/1650.

The question remains, should we convert those pointers into pretty hex numbers in `parseArgs()` ?

Reply:
You're also missing a line break before return here (neat: styling).

Reply:
pointers are "automatically" printed as hex in case parse-args was chosen:

```
func (t *Manager) parseArgs(event *trace.Event) error {
	for i := range event.Args {
		if ptr, isUintptr := event.Args[i].Value.(uintptr); isUintptr {
			event.Args[i].Value = "0x" + strconv.FormatUint(uint64(ptr), 16)
		}
	}

        ...
}
```